pkg/globalnet/controllers/ipam/ipam.go

```diff
@@ -0,0 +1,308 @@
+package ipam
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+func NewController(spec *SubmarinerIpamControllerSpecification, config *InformerConfigStruct) (*Controller, error) {
+	exclusionMap := make(map[string]bool)
+	for _, v := range spec.ExcludeNS {
+		exclusionMap[v] = true
+	}
+	pool, err := NewIpPool(spec.GlobalCIDR)
```

Comment:
Was saving that for later as use case for multiple CIDRs is not clear yet. I was thinking it will likely  be a key:value something like "services:CIDR1,pods:CIDR2" in which case it will become a map. Will convert when have clarity on how exactly we're going to use it.

Reply:
The way we defined it is an array of strings with the CIDRs. No need for key value since we are not going to use it like that for now. I think this needs to be changed ASAP, it can be done in a f-up patch though.

Reply:
Agree with @mpeterson. No need for key/value pairs.